### PR TITLE
Fix doctests in cocotb.binary, and run them as part of the regression test

### PR DESCRIPTION
**Pull request by eric-wieser**

### CODE REVIEW

**Comment by ktbarrett** [comment]
This might be overboard, but is there a way to iterate over all the doctests and build each test into a cocotb test?

**Comment by eric-wieser** [comment]
That's definitely possible using the advanced doctest API, but wasn't something I wanted to dive into.

While it would be cool, it would ultimately be a waste of time if we added support for running `unittest.TestSuite` objects, for which doctests already provide hooks.

File: tests/test_cases/test_cocotb/test_doctests.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import doctest
```
Comment (cmarqu) [inline tests/test_cases/test_cocotb/test_doctests.py]: This should get the SPDX header.

Comment (eric-wieser) [reply tests/test_cases/test_cocotb/test_doctests.py]: Can you remind me what that is?

Comment (cmarqu) [reply tests/test_cases/test_cocotb/test_doctests.py]: Sorry, yes, this one :)

**Comment by eric-wieser** [comment]
Seems that travis decided not to run on this...

**Pull request merged**